index.html

```diff
@@ -628,30 +628,21 @@
           <li>Return <var>acceptPromise</var> and perform the remaining steps
           <a>in parallel</a>.
           </li>
-          <li>For each <var>paymentMethod</var> in
```

Comment:
You're removing the part of the current document that deals with `paymentMethod` matching. The algorithm I thumbnailed in [issue 96](https://github.com/w3c/webpayments-payment-apps-api/issues/96) was intended to supplement `paymentMethod` matching rather than replace it. Basically, you need to put the *filter payments options algorithm* into a loop that evaluates it once for each `paymentMethod` in the request, and only those options that match the `paymentMethod` should be evaluated on each iteration.